### DISCUSSION TITLE
Is it possible to add residue-type bias and make it ignore specific residue types in ESM inverse folding model?

### DISCUSSION THREAD

**Discussion by yakomaxa** [facebookresearch/esm#446] [General] [answered]
My question is about how to control ESM inverse folding model better.
 
I would like to use ESM inverse folding model to re-design certain type of enzymes, but I don't want to introduce specific residue type such as CYS and HIS for experimental reasons. In addition, I want to make the inverse folding model prefer (or disfavor) for specific residue types for various sites. 

Are these things possible for ESM inverse folding model?

**Comment by tomsercu** [comment, marked as answer]
At sampling time you could easily suppress certain residues (there is some discussion about that here in discussion & issues).
Similar idea for biasing the p(xi | x<i) towards certain residues, you could easily add a bias to the log probabilities.

**Reply by yakomaxa** [reply]
Thank you for replying. 
I looked into the inverse folding code and found what to modify.
